src/SymfonyCache/UserContextSubscriber.php

```diff
@@ -130,7 +130,10 @@ protected function cleanupHashLookupRequest(Request $hashLookupRequest, Request
                 $hashLookupRequest->cookies->set($name, $value);
             }
         }
-        $hashLookupRequest->headers->set('Cookie', http_build_query($sessionIds, '', '; '));
+
+        if (count($sessionIds) > 0) {
+            $hashLookupRequest->headers->set('Cookie', http_build_query($sessionIds, '', '; '));
+        }
```

Comment:
@ddeboer do you agree that we should do an 
    } else {
         $hashLookupRequest->headers->remove('Cookie');
    }